__tests__/app.test.js

```diff
@@ -89,4 +89,39 @@ describe("/api/articles/:article_id", () => {
             expect(response.body.msg).toBe("Bad request")
         })
     })
-})
\ No newline at end of file
+})
+describe("GET /api/articles", () => {
+    test("should respond with an array of all article objects", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body: { articles } }) => {
+            expect(articles).toHaveLength(13)
+            articles.forEach((article) => {
+                 expect(Object.keys(article)).toIncludeSameMembers([
+                    "author",
+                    "title",
+                    "article_id",
+                    "topic",
+                    "created_at",
+                    "votes",
+                    "article_img_url",
+                    "comment_count",
+                ])
+            })
+        })
+    })
+    test("should sort the articles by date in descending order", () => {
+        return request(app)
+          .get("/api/articles")
+          .expect(200)
+          .then(({ body: { articles } }) => {
+            expect(articles).toBeSortedBy("created_at", { descending: true })
+        })
+    })
+    test('Status code 404 for bad route',()=>{
+        return request(app)
+        .get('/api/article')
+        .expect(404)
+    })
```

Comment:
This 404 is a generic error test and shouldn't be in the describe block for this ticket as it doesn't directly relate to /api/articles. The bad route could be /api/carrots, aslong as the endpoint doesn't exist.